config/session.php

```diff
@@ -163,10 +163,14 @@
     | even if the user follows a link to another website. Lax cookies are
     | only sent with a top-level get request.
     |
-    | Supported: "lax", "strict"
+    | First-party cookies that match the domain, be set to 'Strict' or 'Lax'.
+    |
+    | Third-party cookies from other domains, be set to 'None' and need
+    | the Secure set to true.
+    |
+    | Supported: "Lax", "Strict" and "None"
```

Comment:
Can you explain what each mode does?

Reply:
@LukeTowers see this article of [MDN](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Headers/Set-Cookie/SameSite)

Reply:
Can you explain it to me in the context of an admin trying to figure out what value they need to set for their specific site?

Reply:
@LukeTowers I see what you are saying (let me try and write a clearer description aimed at a clueless person looking at it for the first time). Please review this and I will try and create a commit for an updated description:

> This option determines how your cookies behave when cross-site requests take place and can be used to mitigate CSRF attacks.
>
> Cookies that match the domain of the current site, i.e. what's displayed in the browser's address bar, are referred to as first-party cookies. Similarly, cookies from domains other than the current site are referred to as third-party cookies.
>
> Cookies without a `SameSite` attribute will be treated as `SameSite=Lax`, meaning the default behaviour will be to restrict cookies to first party contexts only.
>
> Cookies for cross-site usage must specify `SameSite=None` and  `Secure=true` to enable inclusion in third party context.
>
> `Lax` - Cookies are allowed to be sent with top-level navigations and will be sent along with GET request initiated by third party website. This is the default value in modern browsers.
>
> `Strict` - Cookies will only be sent in a first-party context and not be sent along with requests initiated by third party websites.
>
> Supported: "Lax", "Strict" and "None"

I think this is a bit clearer and more custom to October CMS config settings.

Reply:
The branch name is not there? So, if you need the description changing I would have to create a fresh new pr.

Reply:
Okay, so if I understand it correctly the following is what happens?

If you are on example.com and you make a request to example.com, both Strict & Lax cookies belonging to example.com will be attached to that request
If you are on evil.net and you make a request to example.com then only Lax cookies belonging to example.com will be sent? Strict cookies made in this case will not be sent?

Also, how strict are the browsers when looking at things like subdomains when factoring in if a strict cookie will be removed from the request or not?

Reply:
> If you are on example.com and you make a request to example.com, both Strict & Lax cookies belonging to example.com will be attached to that request
> If you are on evil.net and you make a request to example.com then only Lax cookies belonging to example.com will be sent? Strict cookies made in this case will not be sent?

Yes correct.

> Also, how strict are the browsers when looking at things like subdomains when factoring in if a strict cookie will be removed from the request or not?

Browsers will treat the main domain and subdomains exactly the same, they use the same setup as the cross-origin api rules. The main domain and the subdomain (on the same website) will be under first-party and not third-party cookies.

p.s. my above description I removed the part:

> By default, we do not enable this as other CSRF protection services are in place.

Because browsers have said using the SameSite you do not need to use CSRF anymore (but it can't hurt to still use both to have backwards compatibility).

Let me know if I need to re-submit a ne pr or not?

Reply:
To fully answer your question, its best you read about COOP and COEP, see this short video: https://www.youtube.com/watch?v=XLNJYhjA-0c

Reply:
@ayumi-cloud yes, please resubmit a new PR.